### PR TITLE
ci: compile everything under /test in release CI

### PR DESCRIPTION
**Pull request by mattklein123**
This covers tools, benchmarks, etc. It will at least make sure
things don't get stale that aren't tests.

*Risk Level*: None
*Testing*: N/A
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
@htuch @jmarantz I'm not sure if this is the best way of handling this, but I'm pretty sure we are not covering the non-test binaries in CI currently which seems bad.

File: ci/do_ci.sh [hunk 1]
```diff
@@ -49,6 +49,10 @@ if [[ "$1" == "bazel.release" ]]; then
   echo "bazel release build with tests..."
   bazel_release_binary_build
   echo "Testing..."
+  # We have various test binaries in the test directory such as tools, benchmarks, etc. We
+  # run a build pass to make sure they compile.
+  bazel --batch build ${BAZEL_BUILD_OPTIONS} -c opt //test/...
```
Comment (jmarantz) [inline ci/do_ci.sh]: should we also build //source/... ?  Actually while we are at it, what about //include/... ?

Comment (htuch) [reply ci/do_ci.sh]: I think we probably have almost everything in `//source` and `//include` already captured implicitly by `//test`, but I agree that it doesn't hurt to add them here, Bazel won't do extra work.

**Comment by mattklein123** [comment]
updated per comments

**Pull request merged**